torch/lib/TH/THStorage.c

```diff
@@ -104,6 +104,49 @@ TH_API int THLongStorage_inferSize2(THLongStorage *output, long *sizesA, long di
   return 0;
 }
 
+TH_API int THLongStorage_inferSizeN(THLongStorage *output, int n, long **sizes, long *dims, int raiseErrors) {
```

Comment:
Given n tensors with given sizes and dims, broadcast all of them to a single size, which is recorded in output.